include/eve/detail/overload.hpp

```diff
@@ -14,59 +14,88 @@
     struct  callable_object<tag::TAG, Dummy>                                                       \
     {                                                                                              \
-      using tag_type = tag::TAG;                                                                   \
+      using tag_type    = tag::TAG;                                                                \
+                                                                                                   \
+      template<typename Arg, typename... Args>                                                     \
+      requires  (   tag_dispatchable<tag_type,Arg,Args...>                                         \
+                ||  supports_ ## TAG<Arg,Args...>                                                  \
+                )                                                                                  \
+      static EVE_FORCEINLINE constexpr auto call(Arg&& d, Args &&... args)  noexcept               \
+      {                                                                                            \
+        if constexpr( decorator<std::decay_t<Arg>> )                                               \
+        {                                                                                          \
+          check ( delay_t{}, ::eve::detail::types<std::decay_t<Arg>,tag_type>{},                   \
+                  std::forward<Args>(args)...                                                      \
+                );                                                                                 \
+        }                                                                                          \
+        else                                                                                       \
+        {                                                                                          \
+          check ( delay_t{}, ::eve::detail::types<tag_type>{}, std::forward<Arg>(d),               \
+                  std::forward<Args>(args)...                                                      \
+                );                                                                                 \
+        }                                                                                          \
+                                                                                                   \
+        if constexpr( tag_dispatchable<tag_type,Arg,Args...> )                                     \
+        {                                                                                          \
+          return dispatcher ( tag_type{}                                                           \
+                            , std::forward<Arg>(d), std::forward<Args>(args)...                    \
+                            );                                                                     \
```

Comment:
I suspect  that you can just use `adl` here directly:

```
return tag_dispatch(tag_type{}, , std::forward<Arg>(d), std::forward<Args>(args)..);
```

Then dispatcher goes away. 

No need to change it, just saying.